File: src/infrastructure/repository/strategies/repository_strategy.py
from domain.abstracts.base_domain import BaseDomain
from infrastructure.servers.abstracts.base_server import BaseServer
from infrastructure.repository.strategies.abstracts.base_repository_strategy import BaseRepositoryStrategy

class RepositoryStrategy(BaseRepositoryStrategy):

    def __init__(self, server: BaseServer, domain_cls: BaseDomain):
        self._server = server
        self._domain_cls = domain_cls

    def build_domain(self) -> BaseDomain:
        return self._domain_cls

    @property
    def server(self) -> BaseServer:
        return self._server

File: angular-app/src/app/core/infrastructure/repository/strategies/repository-strategy.ts
import { BaseDomain } from '../../../../domain/abstracts/base-domain';
import { BaseServer } from '../../../servers/abstracts/base-server';
import { Builder } from '../../builder';
import { BaseRepositoryStrategy } from './abstracts/base-repository-strategy';

export class RepositoryStrategy implements BaseRepositoryStrategy {
  constructor(
    private _domain: BaseDomain,
    private _builder: Builder
  ) {}

  buildDomain(): BaseDomain {
    return this._domain;
  }

  get server(): BaseServer {
    return this._builder.server;
  }
}
